lib/src/piece/list.dart

```diff
@@ -182,6 +182,39 @@ class ListPiece extends Piece {
 
     if (_after case var after?) callback(after);
   }
+
+  @override
+  State? fixedStateForPageWidth(int pageWidth) {
+    var totalLength = 0;
+    if (_before case var before?) {
+      // A newline in the opening bracket (like a line comment after the
+      // bracket) forces the list to split.
+      if (before.containsNewline) return State.split;
+      totalLength += before.totalCharacters;
+    }
+
+    for (var element in _elements) {
+      // Elements that can be block arguments won't necessarily force the list
+      // to split.
+      if (element.allowNewlines) continue;
+
+      if (element.containsNewline) return State.split;
```

Comment:
Is it possible for the element to contain a newline if it doesn't allow newlines? Maybe I'm reading this wrong.

Reply:
Yeah, for example:

```dart
function(arg, // comment
);
```

Here, the `arg, // comment` will have `allowNewlines` false because it isn't a kind of expression that supports block formatting (collection literal, etc.)

But it will also have `containsNewline` true because of the `// comment` hanging of the end of the element.

So this is a case where we know the list will always split because there's a newline in an argument that doesn't allow a newline when the list is unsplit.

Maybe it's confusing because of the name. The `allowNewlines` field on `ListElementPiece` means "allow newlines *when the list is unsplit*". Renamed to `allowNewlinesWhenUnsplit` to make that clearer. :)